crates/sol-type-parser/src/error.rs

```diff
@@ -57,12 +57,12 @@ impl Error {
     #[inline(never)]
     #[cold]
     fn new_(s: &str, e: &dyn fmt::Display) -> Self {
-        Self(Repr(format!("{s}{e}")))
+        Self(Repr(format!("{s}{e}").into_boxed_str()))
     }
 }
 
 #[derive(Clone, PartialEq, Eq)]
-struct Repr(String);
+struct Repr(Box<str>);
```

Comment:
why do we need this in practice?

Reply:
Not needed really, just want to abstract it away from the main struct. If you mean `Box<str>`, it's one word smaller than `String` (24 -> 16)